chalk-solve/src/clauses.rs

```diff
@@ -377,6 +384,57 @@ fn program_clauses_that_could_match<I: Interner>(
     Ok(clauses)
 }
 
+/// Adds the clause:
+///
+/// forall<T1, T2> Normalize(<T1 as Trait>::Item -> T2)
+///     :- Implemented(T1: Trait), Compatible, DownstreamType(T1), CannotProve
+fn push_clauses_for_compatible_normalize<I: Interner>(
+    builder: &mut ClauseBuilder<'_, I>,
+    interner: &I,
+    trait_id: TraitId<I>,
+    associated_ty_id: AssocTypeId<I>,
+) {
+    // FIXME: Simplify?
+    builder.push_binders(
```

Comment:
Building this clause is long and cumbersome. Is there a better way? Most clauses seem to re-use binders, but I'm not sure if that is possible here.